DQMOffline/Trigger/plugins/TopMonitor.cc

```diff
@@ -574,17 +576,35 @@ void TopMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup
       edm::LogWarning("TopMonitor") << "Electron handle not valid \n";
       return;
   }
+  
+  //ATHER                                                                                                                                                                                                                       
+  edm::Handle<edm::ValueMap<bool> > eleIDHandle;
+  iEvent.getByToken(elecIDToken_,  eleIDHandle);
+  if (!eleIDHandle.isValid()){
+    edm::LogWarning("TopMonitor") << "Electron ID handle not valid \n";
+    return;
+  }
+
+
   std::vector<reco::GsfElectron> electrons;
   if (nelectrons_>0){
-      if ( eleHandle->size() < nelectrons_ ) return;
-      for ( auto const & e : *eleHandle ) {
-          if (eleSelection_(e)) electrons.push_back(e);
-          //Suvankar
-          if ( usePVcuts_ &&
-               (std::fabs(e.gsfTrack()->dxy(pv->position())) >= lepPVcuts_.dxy || std::fabs(e.gsfTrack()->dz(pv->position())) >= lepPVcuts_.dz) ) continue;
-      }
-      if ( electrons.size() < nelectrons_ ) return;
+    if ( eleHandle->size() < nelectrons_ ) return;
+    size_t index=0;                          //ATHER
+    for ( auto const & e : *eleHandle ) {
+      const auto el = eleHandle->ptrAt(index);  //ATHER                                                                                                                                                                                   
+      bool pass_id = (*eleIDHandle)[el];   //ATHER                                                                                                                                                                                         
+      if (eleSelection_(e) && pass_id) electrons.push_back(e); //ATHER
+      //Suvankar
+      if ( usePVcuts_ &&
+	   (std::fabs(e.gsfTrack()->dxy(pv->position())) >= lepPVcuts_.dxy || std::fabs(e.gsfTrack()->dz(pv->position())) >= lepPVcuts_.dz) ) continue;
+      index++;  //ATHER
```

Comment:
are you sure you want this line here ?
if the condition in line 598 is not satisfied, your index will not be incremented !

in addition, do you really need to make use of the index ?

Reply:
You are rite. It should go before line 596 (I am going to update it). And yes index is needed for reading entires from elecID corresponding to electron.